src/main/java/com/scalefocus/blog_api/service/impl/BlogServiceImpl.java

```diff
@@ -86,9 +96,41 @@ public List<BlogDto> getBlogsByTagName(String tagName) {
     @Override
     public List<SimplifiedBlogResponse> getSimplifiedBlogs() {
         List<Blog> blogs = blogRepository.findAll();
-        return blogs.stream().map(blog -> new SimplifiedBlogResponse(blog.getTitle(),blog.getText())).toList();
+        return blogs.stream().map(blog -> new SimplifiedBlogResponse(blog.getTitle(), blog.getText())).toList();
+    }
+
+    @Override
+    public UserResponse deleteUserBlog(Long blogId, Long userId) {
+        User user = userRepository.findById(userId).orElseThrow(() -> new ResourceNotFound("User Not Found"));
+        Blog blog = user.getBlogList().stream()
```

Comment:
there should be simplier solution to this. Think of a method to fetch blog by userName & blog id, and simply delete it if exist. Isnt this going to be much easier ?

Reply:
I thought that user has to get all the blogs of all the users. I changed it to get blogs with username.

Reply:
I got the point now. I changed it to get blog by username and delete it.